apps/passport-client/components/shared/PCDArgs.tsx

```diff
@@ -45,102 +44,142 @@ type ArgSetter = (argName: string, value: any) => void;
 export function PCDArgs<T extends PCDPackage>({
   args,
   setArgs,
-  pcdCollection
+  options
 }: {
   args: ArgsOf<T>;
   setArgs: React.Dispatch<React.SetStateAction<ArgsOf<T>>>;
-  pcdCollection: PCDCollection;
+  options?: ArgsDisplayOptions<ArgsOf<T>>;
 }) {
-  const entries = Object.entries(args);
-
-  const setArg = React.useCallback(
-    (argName: string, value: any) => {
-      setArgs((args) => ({
-        ...args,
-        [argName]: {
-          ...args[argName],
-          value
-        }
-      }));
-    },
-    [setArgs]
+  const [showAll, setShowAll] = useState(false);
+  const [visible, hidden] = _.partition(
+    Object.entries(args),
+    ([key]) => options?.[key]?.defaultVisible ?? true
   );
 
   return (
     <ArgsContainer>
-      {entries.map(([key, value], i) => (
+      {visible.map(([key, value]) => (
         <ArgInput
-          pcdCollection={pcdCollection}
-          key={i}
+          key={key}
           argName={key}
           arg={value as any}
-          setArg={setArg}
+          setArgs={setArgs}
+          defaultArg={options?.[key]}
         />
       ))}
+      {hidden.length > 0 && (
+        <>
+          <ShowMoreButton
+            showAll={showAll}
+            onClick={() => setShowAll((showAll) => !showAll)}
+            size="sm"
+          >
+            {showAll ? "▼ Hide" : "► Show"} {hidden.length} more inputs
+          </ShowMoreButton>
+          {showAll &&
```

Comment:
unless I 'show all', I think the arguments that are hidden never actually get initialized with a value. eg. when i go to consumer client and request a zk ticket pcd proof, and don't expand the hidden items, i am not able to generate a proof because the pcd args did not select my one and only identity. this is a bug.


<img width="456" alt="Screenshot 2023-10-09 at 6 13 30 PM" src="https://github.com/proofcarryingdata/zupass/assets/2636237/9597df7c-1764-474b-8735-7ec33eefe175">

Reply:
oops. good catch